case-studies/n-body/entt/main.cpp

```diff
@@ -86,33 +84,74 @@ void update(entt::registry& registry) {
     movement(registry);
 }
 
-int main() {
-    entt::registry registry;
-
-    std::default_random_engine rnd;
-    std::uniform_real_distribution<float> radius_distribution(0, 1);
-    std::uniform_real_distribution<float> position_distribution(-5, 5);
-
-    for (int i = 0; i < BODY_COUNT; i++) {
-        const auto entity = registry.create();
-        auto radius = radius_distribution(rnd);
-        registry.emplace<Position>(entity, Eigen::Vector3f(position_distribution(rnd), position_distribution(rnd), position_distribution(rnd)));
-        registry.emplace<Velocity>(entity, Eigen::Vector3f::Zero());
-        registry.emplace<Acceleration>(entity, Eigen::Vector3f::Zero());
-        registry.emplace<Mass>(entity, 1e6f * radius * radius * radius);
-    }
-
-    auto start = std::chrono::high_resolution_clock::now();
+struct Benchmark {
+  int body_count;
+  int ticks_per_measurement;
+  int measurement_count;
+
+  Benchmark(int body_count, int simulated_ticks, int measurement_count = 10) : body_count(body_count), measurement_count(measurement_count) {
+      ticks_per_measurement = simulated_ticks / measurement_count;
+  }
+
+  void run() const {
+      entt::registry registry;
+
+      std::default_random_engine rnd;
+      std::uniform_real_distribution<float> radius_distribution(0, 1);
+      std::uniform_real_distribution<float> position_distribution(-5, 5);
+
+      for (int i = 0; i < body_count; i++) {
+          const auto entity = registry.create();
+          auto radius = radius_distribution(rnd);
+          registry.emplace<Position>(entity, Eigen::Vector3f(position_distribution(rnd), position_distribution(rnd), position_distribution(rnd)));
+          registry.emplace<Velocity>(entity, Eigen::Vector3f::Zero());
+          registry.emplace<Acceleration>(entity, Eigen::Vector3f::Zero());
+          registry.emplace<Mass>(entity, 1e6f * radius * radius * radius);
+      }
+      
+      std::vector<std::chrono::duration<double>> measurements;
+
+      for (int i = 0; i < measurement_count; i++) {
+          auto start = std::chrono::high_resolution_clock::now();
+          for (int j = 0; j < ticks_per_measurement; j++) {
+              update(registry);
+          }
+          auto stop = std::chrono::high_resolution_clock::now();
+          measurements.emplace_back(stop - start);
+          std::cout << "." << std::flush;
+      }
+      
+      double mean = 0;
+      for (auto& measurement : measurements) {
+          mean += measurement.count();
+      }
+      mean /= measurement_count;
+      
+      double variance = 0;
+      for (auto& measurement : measurements) {
+          auto val = measurement.count();
+          variance += (val - mean) * (val - mean);
+      }
+      auto standard_deviation  = sqrt(variance);
```

Comment:
Are you sure this computation for variance is correct? I think you might be missing the division by `n - 1`. I'm comparing against [this two-pass algorithm](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Two-pass_algorithm).